### PR TITLE
Fix cmake issues

### PR DESCRIPTION
**Pull request by luator**
Fix several issues in CmakeLists.txt that broke the build with `catkin build` or `catkin_make_isolated`.
See commits for details

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +25,7 @@ catkin_package(
   LIBRARIES ${LIBRARY_NAME}
   INCLUDE_DIRS include
   CATKIN_DEPENDS mpi_cmake_modules
+  DEPENDS Boost
```
Comment (MaximilienNaveau) [inline CMakeLists.txt:28]: Don't we need to add a package.xml info as well on this? I think there is a special xml stuff for the system dependencies.

Comment (luator) [reply CMakeLists.txt:28]: You are right, there should be a `<depend>boost</depend>` in the package.xml, I'll add it.

**Comment by luator** [comment]
@MaximilienNaveau @vincentberenz Can this be merged?

**Review by MaximilienNaveau — APPROVED** [review]
I think this is good, thanks!

**Comment by luator** [comment]
@vincentberenz Please review as well, this needs two approvals

**Comment by MaximilienNaveau** [comment]
I took the freedom of merging this.

**Pull request merged**